File: src/lib/pjecalc/core/comum/validators/min-validator.ts
/**
 * PJe-Calc v2.15.1 — MinValidator
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.MinValidator
 */
import Decimal from 'decimal.js';
import { CustomValidator } from './custom-validator';
import type { Min_ } from './min';
import type { ValidatorContext } from './validator-context';

export class MinValidator extends CustomValidator<Min_> {
  getMessage(): string { return this.spec.message; }
  getGroups(): number[] { return this.spec.groups; }
  protected getCondition(): string { return this.spec.condition; }

  isValid(context: ValidatorContext): boolean {
    const value = context.getValue();
    if (value === null || value === undefined) return true;
    const min = new Decimal(this.spec.value);
    if (typeof value === 'string') {
      try {
        return new Decimal(value).greaterThanOrEqualTo(min.floor());
      } catch {
        return false;
      }
    }
    if (value instanceof Decimal) {
      return value.greaterThanOrEqualTo(min);
    }
    if (typeof value === 'number') {
      return value >= min.toNumber();
    }
    if (typeof value === 'bigint') {
      return value >= BigInt(min.floor().toString());
    }
    return false;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/MinValidator.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.base.comum.validadores.CustomValidator;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.Min;
import java.math.BigDecimal;
import java.math.BigInteger;

public class MinValidator
extends CustomValidator<Min> {
    @Override
    public Object getParameter(String key) {
        return null;
    }

    @Override
    public String getMessage() {
        return ((Min)this.getAnnotation()).message();
    }

    @Override
    public byte[] getGroups() {
        return ((Min)this.getAnnotation()).groups();
    }

    @Override
    public boolean isValid(ValidatorContext context) {
        Object value = context.getValue();
        if (value == null) {
            return true;
        }
        BigDecimal min = new BigDecimal(((Min)this.getAnnotation()).value());
        if (value instanceof String) {
            try {
                return new BigDecimal((String)value).compareTo(BigDecimal.valueOf(min.intValue())) >= 0;
            }
            catch (NumberFormatException nfe) {
                return false;
            }
        }
        if (value instanceof Double || value instanceof Float) {
            double dv = ((Number)value).doubleValue();
            return dv >= min.doubleValue();
        }
        if (value instanceof BigInteger) {
            return ((BigInteger)value).compareTo(BigInteger.valueOf(min.intValue())) >= 0;
        }
        if (value instanceof BigDecimal) {
            return ((BigDecimal)value).compareTo(min) >= 0;
        }
        if (value instanceof Number) {
            long lv = ((Number)value).longValue();
            return lv >= min.longValue();
        }
        return false;
    }
}

